### PR TITLE
update citations to match the linked whitepaper

### PR DESCRIPTION
**Pull request by djwadhwa**
Description: There are a few inline documentation citations that are not correct. I have used the associated whitepaper links provided to update the citation (including author and year)

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [X] Documentation is updated (if required)

### CODE REVIEW

File: ignite/metrics/gan/fid.py [hunk 1]
```diff
@@ -62,11 +62,11 @@ class FID(_BaseInceptionMetric):
     where :math:`\mu_1` and :math:`\sigma_1` refer to the mean and covariance of the train data and
     :math:`\mu_2` and :math:`\sigma_2` refer to the mean and covariance of the test data.
 
-    More details can be found in `Heusel et al. 2002`__
+    More details can be found in `Heusel et al. 2018`__
```
Comment (vfdev-5) [inline ignite/metrics/gan/fid.py:65]: arxiv references it as 2017, let's put 2017 even if the latest revision version is 2018

File: ignite/metrics/gan/fid.py [hunk 2]
```diff
@@ -62,11 +62,11 @@ class FID(_BaseInceptionMetric):
...
 
     __ https://arxiv.org/pdf/1706.08500.pdf
 
-    In addition, a faster and online computation approach can be found in `Chen et al. 2014`__
+    In addition, a faster and online computation approach can be found in `Mathiasen et al. 2021`__
```
Comment (vfdev-5) [inline ignite/metrics/gan/fid.py:69]: Same here, 2020 instead of 2021

**Comment by djwadhwa** [comment]
Thank you for pointing these out @vfdev-5! I have updated the documentation to align with the recommended changes.

**Review by vfdev-5 — APPROVED** [review]
Thanks @djwadhwa , lgtm!

**Pull request merged**